File: packages/shared/src/pub.ts
import * as v from 'valibot';

/** 앱 부트스트랩 시 클라이언트가 읽는 공개 설정 — 유저별로 달라지지 않는다(공유 캐시 대상). */
export const appConfigResponseSchema = v.object({
  version: v.string(),
  minSupportedAppVersion: v.string(),
  features: v.object({
    projects: v.boolean(),
    timeline: v.boolean(),
  }),
});

export type AppConfigResponse = v.InferOutput<typeof appConfigResponseSchema>;

File: apps/bff-rs/crates/contract/src/schemas/pub_.rs
//! Mirror of `packages/shared/src/pub.ts`.

use crate::ir::*;

pub fn module() -> Module {
    Module {
        name: "pub",
        decls: vec![schema_doc(
            "appConfigResponseSchema",
            "앱 부트스트랩 시 클라이언트가 읽는 공개 설정 — 유저별로 달라지지 않는다(공유 캐시 대상).",
            "AppConfigResponse",
            object(vec![
                field("version", string()),
                field("minSupportedAppVersion", string()),
                field(
                    "features",
                    object(vec![
                        field("projects", boolean()),
                        field("timeline", boolean()),
                    ]),
                ),
            ]),
        )],
    }
}
